Jetsurvey/app/src/main/java/com/example/compose/jetsurvey/survey/SurveyQuestions.kt

```diff
@@ -258,27 +270,114 @@ private fun ActionQuestion(
     onAction: (Int, SurveyActionType) -> Unit,
     modifier: Modifier = Modifier
 ) {
-    Button(
-        onClick = { onAction(questionId, possibleAnswer.actionType) },
-        modifier = modifier.padding(vertical = 20.dp)
-    ) {
-        Text(text = stringResource(id = possibleAnswer.label))
+    when (possibleAnswer.actionType) {
+        SurveyActionType.PICK_DATE -> {
+            DateQuestion(
+                questionId = questionId,
+                answerLabel = possibleAnswer.label,
+                answer = answer,
+                onAction = onAction,
+                modifier = modifier
+            )
+        }
+        SurveyActionType.TAKE_PHOTO -> {
+            PhotoQuestion(
+                questionId = questionId,
+                answer = answer,
+                onAction = onAction,
+                modifier = modifier
+            )
+        }
+        SurveyActionType.SELECT_CONTACT -> TODO()
     }
-    if (answer != null) {
-        when (answer.result) {
-            is SurveyActionResult.Date -> {
-                Text(
-                    text = stringResource(R.string.selected_date, answer.result.date),
-                    style = MaterialTheme.typography.h4,
-                    modifier = Modifier.padding(vertical = 20.dp)
+}
+
+@Composable
+private fun PhotoQuestion(
+    questionId: Int,
+    answer: Answer.Action?,
+    onAction: (Int, SurveyActionType) -> Unit,
+    modifier: Modifier = Modifier
+) {
+    val resource = if (answer != null) {
+        Icons.Filled.SwapHoriz
+    } else {
+        Icons.Filled.AddAPhoto
+    }
+    val label = if (answer != null) {
+        R.string.retake_photo
+    } else {
+        R.string.add_photo
+    }
+    OutlinedButton(
+        onClick = { onAction(questionId, SurveyActionType.TAKE_PHOTO) },
+        modifier = modifier,
+        contentPadding = PaddingValues()
+    ) {
+        Column {
+            if (answer != null && answer.result is SurveyActionResult.Photo) {
+                CoilImage(
+                    data = answer.result.uri,
+                    modifier = Modifier.fillMaxSize()
                 )
+            } else {
+                PhotoDefaultImage(modifier = Modifier.padding(horizontal = 86.dp, vertical = 74.dp))
+            }
+            Row(
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .wrapContentSize(Alignment.BottomCenter)
+                    .padding(vertical = 26.dp),
+                verticalAlignment = Alignment.Bottom
```

Comment:
The alignment of the image and the 'Retake Photo' text looks off, does centering look better?